### PR TITLE
Disable the display of notes Dialog after Ctrl+Shift+N for artificial…

### LINKED ISSUE #6308: Crash on Ctrl+Shift+N for artificial commits
## Current behaviour

Selecting Working Directory and hitting Ctrl+Shift+N crashes gitextensions.

## Expected behaviour

No crash and either add note or not add note dialog.

## Steps to reproduce

In Commit tree
Select working directory
Press Ctrl+Shift+N

Crash dialog comes up.

## Did this work in previous version of GitExtensions

Not sure.

## Environment

- GitExtensions version: 3.0.2
- GIT version: 2.21.0
- OS version: Windows 10 1809
- .NET version: Not sure

### PR DESCRIPTION
**Pull request by ivangrek**
Fixes #6308

## Test methodology 

- Pressed the key combination Ctrl+Shift+N

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Comment by gerhardol** [comment]
Can you please sign first, preferable as a separate PR?
https://github.com/gitextensions/gitextensions/blob/master/contributors.txt

(wiki must be updated)

File: GitUI/CommandsDialogs/FormBrowse.cs [hunk 1]
```diff
@@ -2154,7 +2154,14 @@ internal Keys GetShortcutKeys(Command cmd)
 
         private void AddNotes()
         {
-            Module.EditNotes(RevisionGrid.GetSelectedRevisions().FirstOrDefault()?.ObjectId);
+            var objectId = RevisionGrid.GetSelectedRevisions().FirstOrDefault()?.ObjectId;
+
+            if (objectId == ObjectId.IndexId || objectId == ObjectId.WorkTreeId)
```
Comment (gerhardol) [inline GitUI/CommandsDialogs/FormBrowse.cs]: Should use objectId.IsArtificial

Comment (RussKie) [reply GitUI/CommandsDialogs/FormBrowse.cs]: :warning: `objectId` can be `null`

**Comment by ivangrek** [comment]
In what cases objectId can be null?

**Comment by RussKie** [comment]
> In what cases objectId can be null?

`RevisionGrid.GetSelectedRevisions().FirstOrDefault()?.ObjectId` is a nullable result.

**Comment by ivangrek** [comment]
Maybe it requires an exception call?

**Comment by vbjay** [comment]
Nope. Why use an exception for program flow?

https://docs.microsoft.com/en-us/dotnet/api/system.nullable-1.hasvalue

**Comment by RussKie** [comment]
Thank you

**Pull request merged**